pkg/cmd/repo/garden/garden.go

```diff
@@ -0,0 +1,412 @@
+	opts := GardenOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:    "garden [<repository>]",
+		Short:  "Explore a git repository as a garden",
+		Long:   "Use WASD or vi keys to move. q to quit.",
+		Hidden: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return gardenRun(&opts)
+		},
+	}
+
+	return cmd
+}
+
+func gardenRun(opts *GardenOptions) error {
+	out := opts.IO.Out
+
+	if runtime.GOOS == "windows" {
+		return errors.New("sorry :( this command only works on linux and macos")
+	}
+
+	if !opts.IO.IsStdoutTTY() {
+		return errors.New("must be connected to a terminal")
+	}
+
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	var toView ghrepo.Interface
+	apiClient := api.NewClientFromHTTP(httpClient)
+	if opts.RepoArg == "" {
+		var err error
+		toView, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+	} else {
+		var err error
+		viewURL := opts.RepoArg
+		if !strings.Contains(viewURL, "/") {
+			currentUser, err := api.CurrentLoginName(apiClient, ghinstance.Default())
+			if err != nil {
+				return err
+			}
+			viewURL = currentUser + "/" + viewURL
+		}
+		toView, err = ghrepo.FromFullName(viewURL)
+		if err != nil {
+			return fmt.Errorf("argument error: %w", err)
+		}
+	}
+
+	seed := computeSeed(ghrepo.FullName(toView))
+	rand.Seed(seed)
+
+	termWidth, termHeight, err := utils.TerminalSize(out)
+	if err != nil {
+		return err
+	}
+
+	termWidth -= 10
+	termHeight -= 10
+
+	geo := &Geometry{
+		Width:      termWidth,
+		Height:     termHeight,
+		Repository: toView,
+		// TODO based on number of commits/cells instead of just hardcoding
+		Density: 0.3,
+	}
+
+	maxCommits := geo.Width * geo.Height
+
+	commits, err := getCommits(httpClient, toView, maxCommits)
+	if err != nil {
+		return err
+	}
+	player := &Player{0, 0, utils.Bold("@"), geo, 0}
+
+	garden := plantGarden(commits, geo)
+	clear(opts.IO)
+	drawGarden(out, garden, player)
+
+	// thanks stackoverflow https://stackoverflow.com/a/17278776
+	if runtime.GOOS == "darwin" {
+		_ = exec.Command("stty", "-f", "/dev/tty", "cbreak", "min", "1").Run()
+		_ = exec.Command("stty", "-f", "/dev/tty", "-echo").Run()
+	} else {
+		_ = exec.Command("stty", "-F", "/dev/tty", "cbreak", "min", "1").Run()
+		_ = exec.Command("stty", "-F", "/dev/tty", "-echo").Run()
+	}
+
+	var b []byte = make([]byte, 1)
+	for {
+		_, err := opts.IO.In.Read(b)
+		if err != nil {
+			return err
+		}
+
+		quitting := false
+		switch {
+		case isLeft(b):
+			player.move(DirLeft)
+		case isRight(b):
+			player.move(DirRight)
+		case isUp(b):
+			player.move(DirUp)
+		case isDown(b):
+			player.move(DirDown)
+		case isQuit(b):
+			quitting = true
+		}
+
+		if quitting {
+			break
+		}
+
+		clear(opts.IO)
```

Comment:
I realize that this would add some complexity, but is there a chance you could use cursor positioning to only redraw the 1–2 rows of the garden around the player's current position after a move (plus the status line in the bottom)? Redrawing the whole garden in iTerm2 for me makes the bottom few rows of the garden flicker for me, which is not a huge deal but it does make for a slightly worse experience, especially because it also causes the status line to flicker, which makes the text harder to read while I'm walking around.

Reply:
the flickering is present in all platforms; only redrawing the current rows is a great improvement that i just haven't made time to do.

Reply:
Yeah, I realize how that could delay the whole PR. Let's keep that as a potential improvement for later.

It's not that I feel very strongly that the garden always needs to render super smooth, but I think it would be valuable for us to learn how to position cursor in the terminal in case we ever need it in other commands or while redoing prompts in a more accessible manner.

Reply:
agreed, and it's important to me personally to learn how to do it! I will timebox an hour for it today but otherwise wait until a hack day post GA

Reply:
ok, garden is now redrawing only the characters that need to change (status line, player's character, the cell the player was previously on).